CoreTimelineView/src/au/gov/asd/tac/constellation/views/timeline/TimelineSelectionPlugin.java

```diff
@@ -50,16 +54,32 @@ public TimelineSelectionPlugin(final Set<Integer> vertices, final Set<Integer> t
     @Override
     public void edit(final GraphWriteMethods graph, final PluginInteraction interaction,
             final PluginParameters parameters) throws InterruptedException {
+        final boolean vertexAttributeUndefined = Graph.NOT_FOUND == graph.getAttribute(GraphElementType.VERTEX,
+                VisualConcept.VertexAttribute.SELECTED.getName());
+        final boolean transactionAttributeUndefined = Graph.NOT_FOUND == graph.getAttribute(GraphElementType.TRANSACTION,
+                VisualConcept.TransactionAttribute.SELECTED.getName());
         final int selectedVertexAttrID = VisualConcept.VertexAttribute.SELECTED.ensure(graph);
         final int selectedTransactionAttrID = VisualConcept.TransactionAttribute.SELECTED.ensure(graph);
```

Comment:
I think I need some help understanding this one. It seems a bit odd to me that we're adding a boolean variable for whether the Selected attributes exist and then creating those attributes immediately after

Reply:
This was done to replicate the functionality it was using previously, but I suspect with the old code it was never actually entering that code segment otherwise we would have been getting the ConcurrentModificationException more often.
I was considering fully removing the code block which removes vertices and transaction, as I was not able to replicate a situation that would trigger that code.